lib/ecto/adapters/myxql.ex

```diff
@@ -238,7 +238,7 @@ defmodule Ecto.Adapters.MyXQL do
         {:ok, last_insert_id(key, last_insert_id)}
 
       {:error, err} ->
-        case @conn.to_constraints(err) do
+        case @conn.to_constraints(err, [source: sql]) do
```

Comment:
I saw some tests that extracted multiple tables names, but since we need to catch only insert and update statements it would make sense to simple pass the first table name.But i will double check.

Reply:
Thanks. Using the first table is enough. I have pushed another commit.